### DISCUSSION TITLE
оружие и гаджеты

### DISCUSSION THREAD

**Discussion by Pigzinzspace** [godot-engine-ru/shooter-3d#11] [Ideas]
гаджеты и оружие работают так  у героя есть 10 оружейных слотов, от 1 до 0, каждое оружие подходит в какой то слот, в каждый тип (номер)  слота можно вставить только 1 оружие 
гаджеты это предметы которые надо найти в игре 1 раз и потом они всегда будут с тобой они автоматически используются когда нужны, например ключ от двери  или огнеупорный костюм.

шаблон
**"название оружия "**
слот "номер слота"
"описание как работает оружие" 

**кувалда**
слот 1 
наносит максимальный для мили урон, медленная атака, постоянно даже когда не экипирована уменьшает скорость передвижения игрока 

**цепь**
 слот 1 
 позволяет после атаки еще раз нажать кнопку попытаться сбить цель с ног
 
 **разводной ключ**
 слот 1 
 позволяет откручивать гайки, перекусывать душки замков, некоторые вещи без него не открыть. 

**дробовик**
слот 2
урон на ближней и средней дистанции 1-15 метров, может поражать несколько стоящих вплотную целей 
 
**огнемет**
слот 3
сильный урон по площади на ближней дистанции (1-3 метра) в секторе перед игроком куда повернута струя пламени, цели загораются и продолжают получать периодический урон - гореть.

**химическая ловушка**
слот 6
делается из пакетов со свежей кровью и пустых бутылок, считается что бутылки у Героя всегда с собой он получает сразу готовые гранаты, ловушка разбивается или о стену или о зомби и отвлекает внимание зомби от игрока, действует 5 секунд.    

**крио граната**
слот 6
замораживает всех в радиусе 5 метров от падения, 1 секунду не могут двигаться, потом двигаются как в слоумо еще 5 секунд постепенно приходя в норму.

**Comment by alex7850** [comment]
Фишка со слотами интересная, но получается оружия под один слот может быть дофига, предлагаю пока ограничить фантазию слотами от 1 до 5 для большей интуитивности

Также не упомянуто наличие инвентаря помимо слотов. Я правильно понял, что его не будет и игроку придется выбирать какое оружие для первого слота (например) подобрать или оставить? По мне так хорошая идея, если много за нужным оружием бекранить не придется

Цеп штука классная, можно ей не только врагов дергать, но и всякие бочки, ящики, головоломки.
Конечно, графически очень абстрактно скорее всего выйдет анимацию замаха сделать

**Reply by Pigzinzspace** [reply]
да именно так ты можешь таскать только одно оружие для каждого слота, на слоты с 6 по 0 я хотел повесить всякие прибамбасы типо парашюта или глайдера которые нужны раз в 100 лет если они будут так что да так и будет основные слоты 1-5 выделенного инвентаря не будет все квестовые игровые предметы срабатывают автоматом

**Comment by me2beats** [comment]
10 слотов это 10 принципиально разных типов оружия?
Игрок может бегать с несколькими оружиями в одном слоте?

**Reply by Pigzinzspace** [reply]
оружие 5 первых остальное не оружие или очень специфическое оружие
с несколькими в одном слоте бегать нельзя, поднимаешь и автоматически выкидывается старая пушка,  из слота

**Comment by me2beats** [comment]
**граната**

**дробовик**

**автомат**

**узи**?

**пистолет**

**снайперка** 
её реквестовали, но насчёт неё не уверен, миссии вроде с небольшим пространством

**бензопила**?

**лом**

**Reply by Pigzinzspace** [reply]
можно выдавать оружие из категорий случайно, просто по носеру, это будет влиять на порхождение
по моей задумке герой проходит первые 2 миссии с милишным оружием и им же убивает первого босса  

граната мне лично не нравится по тому что это либо чит либо она до смешного мало должна снимать как в  ку2 ​

расклад по оружию такой (это не конкретные позиции просто чтоб было понятно деление по слотам)
1 - слот мили оружие дубинка, газовый ключ, кувалда, цепь
2 - ренжовое оружее короткой дистанции писталет , узи, дробовик, обрез 
3 - специфичесике средства огнемет, гранатомет, ракетница 
4 - автоматические или полуавтоматические нарезные и гладкоствольные винтовки карабины и автоматы 
5 - какое то супер оружие 

у нас вроде никто  пока не против нереалистичных стволов типо алайн бластера или плазмомета так что они тоже могут появиться  в будущем

**Reply by me2beats** [reply]
Гранат просто может мало выдаваться

**Reply by slapin** [reply]
Гранаты это тактическое оружие, под них надо делать дизайн миссии, иначе они легко могут оказаться бесполезным оружием.

**Reply by Pigzinzspace** [reply]
вариант ввести шоковые гранаты мне нравится особенно если их будет пара штук на миссию и еще пара в секретах

**Comment by slapin** [comment]
По слотам это подход из doom/quake это норм но это не значит что 1 слот - 1 оружие, можно ж и несколько повесить, например на 1 Кулаки (дефолт), кастет (больнее), цепь (больнее но дольше и с кулдауном) и тп.

**Reply by Pigzinzspace** [reply]
имеется в виду что ты не можешь таскать все четвертые стволы, у тебя есть автомат хочешь взять винтарь выкинь автомат, а поменять ты их можешь 
просто пока у тебя например дробовик ты будешь находить патроны2 которые подходят к  дробовику, а если возьмешь узи то будешь находить патроны2 для узи

**Reply by slapin** [reply]
Просто необязательно ограничивать настолько сильно, Blood например вполне решал задачу с несколькими видами оружия на слот. С патронами в современных шутерах вопрос решается ящиками где лежат патроны под все что у тебя есть сейчас, если нет особых ограничений конкретных видов оружия (типа ракеты-гранатометы отдельно, стрелковое отдельно, снайперки отдельно).
Можно вообще сделать что в стрелковом оружии патроны назначаются на весь слот сразу.

**Reply by slapin** [reply]
А спецоружие лучше выдавать на ограниченное время (пока патроны не кончатся) и в пределах миссии (не остается на следующую)

**Comment by Einshtein12** [comment]
Предлагаю добавить миниботов (помощников) они могут получать урон, и наносить damage когда игрок находиться на отдалении. Их можно прокачивать и хилить(даже если они мертвы)

**Reply by Pigzinzspace** [reply]
была идея с питомцем (механическим псом) это отсылка ко второму фолауту, но пока не понятно в принципе его можно добавить с пометкой (не основной контент, чтоб за него взялись когда все остальное уже готово)

**Reply by slapin** [reply]
Пес может быть сложно, но какой-нить левитирующий туррель или робот-пылесос не колесах можно было бы сделать напарником. У нас жеж вроде спасение выживших заложено, можно их сделать напарниками?

**Reply by me2beats** [reply]
спасение выживших? вроде не было такого

**Reply by me2beats** [reply]
Левитирующий туррель интересно, ну это как отдельный вид оружия скорее

**Reply by Pigzinzspace** [reply]
не спасение выживших у нас не планируется пока, как то в формат не вписывается слегка 

а почему собака так сложно , просто носится моделька, у нее будет простое поведение, с ней нельзя будет взаимодействовать и отдавать приказы сама будет либо агрить либо станить ближнего к герою монстра 

она даже еще проще робота пылесоса так как не будет стрелять, она будет механическая и не обязательно анатоимческая, как собаки гугл например.
